api/src/test/java/org/openmrs/api/OrderServiceTest.java

```diff
@@ -3413,4 +3413,39 @@ public void shouldSetTheCorrectSortWeightWhenAddingAnOrderWithANegativePosition(
 		expectedException.expectMessage("Cannot add a member which is out of range of the list");
 		secondSavedOrderGroup.addOrder(newOrderWithInvalidPosition, secondSavedOrderGroup.getOrders().size() + 1);
 	}
+        
+        @Test
+	public void shouldInitializeOrderGroupInOrder() {		
+		executeDataSet(ORDER_SET);
+		Encounter encounter = encounterService.getEncounter(3);
+				
+		OrderGroup orderGroup = new OrderGroup();
+		orderGroup.setPatient(encounter.getPatient());
+		orderGroup.setEncounter(encounter);
+		
+		Order firstOrder = new OrderBuilder().withAction(Order.Action.NEW).withPatient(7).withConcept(1000)
+		        .withCareSetting(1).withOrderer(1).withEncounter(3).withDateActivated(new Date()).withOrderType(17)
+		        .withUrgency(Order.Urgency.ON_SCHEDULED_DATE).withScheduledDate(new Date())
+		        .build();
+		
+		Order secondOrder = new OrderBuilder().withAction(Order.Action.NEW).withPatient(7).withConcept(1001)
+		        .withCareSetting(1).withOrderer(1).withEncounter(3).withDateActivated(new Date()).withOrderType(17)
+		        .withUrgency(Order.Urgency.ON_SCHEDULED_DATE).withScheduledDate(new Date())
+		        .build();
+		
+                orderGroup.addOrder(firstOrder, 0);
+		orderGroup.addOrder(secondOrder, 1);
+		                                				
+                Context.getOrderService().saveOrderGroup(orderGroup);
+                
+		OrderGroup savedOrderGroup = Context.getOrderService().getOrderGroupByUuid(orderGroup.getUuid());
+                
+		Context.flushSession();	
```

Comment:
Do you need to flush? In theory you should actually be testing OrderGroup.addOrder method

Reply:
With that said, the test should be in OrderGroupTest class since you will be testing OrderGroup.addOrder, if it doesn't exist you need to add it.